packages/neo-one-client/src/__tests__/Client.test.js

```diff
@@ -0,0 +1,235 @@
+import * as argAssertions from '../args';
+import { UnknownAccountError } from '../errors';
+import { transactions } from '../__data__';
+import createSmartContract from '../sc/createSmartContract';
+
+jest.mock('../sc/createSmartContract');
+
+describe('Client', () => {
+  const id1 = { network: 'net1', address: 'addr1' };
+  const id2 = { network: 'net2', address: 'addr2' };
+  const account1 = {
+    type: 'test1',
+    id: id1,
+    name: 'account1',
+    scriptHash: 'hash160-1',
+    publicKey: 'pukey-1',
+  };
+  const account2 = {
+    type: 'test2',
+    id: id2,
+    name: 'account2',
+    scriptHash: 'hash160-2',
+    publicKey: 'pubkey-2',
+  };
+  const network1 = 'net1';
+  const network2 = 'net2';
+  const provider1 = {
+    type: 'test1',
+    currentAcount$: account1,
+    accounts$: [account1],
+    networks$: [network1],
+  };
+  const provider2 = {
+    type: 'test2',
+    currentAccount$: account2,
+    accounts$: [account2],
+    networks$: [network2],
+  };
+
+  // $FlowFixMe
+  let client = new Client({ test1: provider1, test2: provider2 });
+  // $FlowFixMe
+  let provider = client._selectedProvider$.value;
+  let providers = utils.values(client.providers);
+  beforeEach(() => {
+    // $FlowFixMe
+    client = new Client({ test1: provider1, test2: provider2 });
+    // $FlowFixMe
+    provider = client._selectedProvider$.value;
+    providers = utils.values(client.providers);
+  });
+
+  test('Client constructor throws error on null client', () => {
+    function testError() {
+      client = new Client({});
+    }
+
+    expect(testError).toThrow(new Error('At least one provider is required'));
+  });
+
+  test('Client constructor throws error on mismatched provider type', () => {
+    function testError() {
+      client = new Client({ test: { type: 'invalid' } });
+    }
+
+    expect(testError).toThrow(
+      new Error('Provider keys must be named the same as their type'),
+    );
+  });
+
+  test('get providers', () => {
+    const result = client.providers;
+    // $FlowFixMe
+    expect(result).toEqual(client._providers$.value);
+  });
+
+  test('selectAccount throws error on unknown account', async () => {
+    const accountID = { network: 'fakeNet', address: 'fakeAddr' };
+
+    // $FlowFixMe
+    providers[0].getAccounts = jest.fn(() => providers[0].accounts$);
+    // $FlowFixMe
+    providers[1].getAccounts = jest.fn(() => providers[1].accounts$);
+
+    const result = client.selectAccount(accountID);
+
+    await expect(result).rejects.toEqual(
+      new UnknownAccountError(accountID.address),
+    );
+  });
+
+  test('selectAccount', async () => {
+    const accountID = id2;
+    const expectedProvider = provider2;
+    // $FlowFixMe
+    providers[0].getAccounts = jest.fn(() => providers[0].accounts$);
+    // $FlowFixMe
+    providers[1].getAccounts = jest.fn(() => providers[1].accounts$);
+    // $FlowFixMe
+    providers[1].selectAccount = jest.fn(async () => Promise.resolve({}));
```

Comment:
Returning a resolved promise in an `async` function is redundant. Also, just `Promise.resolve()`.